tsl/src/deparse.c

```diff
@@ -922,3 +922,84 @@ deparse_oid_function_call_coll(Oid funcid, Oid collation, unsigned int num_args,
 
 	return result;
 }
+
+const char *
+deparse_grant_revoke_on_database(Node *node, const char *dbname)
+{
+	GrantStmt *stmt = castNode(GrantStmt, node);
+	ListCell *lc;
+
+	/*
+	   GRANT { { CREATE | CONNECT | TEMPORARY | TEMP } [, ...] | ALL [ PRIVILEGES ] }
+	   ON DATABASE database_name [, ...]
+	   TO role_specification [, ...] [ WITH GRANT OPTION ]
+	   [ GRANTED BY role_specification ]
+
+	   REVOKE [ GRANT OPTION FOR ]
+	   { { CREATE | CONNECT | TEMPORARY | TEMP } [, ...] | ALL [ PRIVILEGES ] }
+	   ON DATABASE database_name [, ...]
+	   FROM role_specification [, ...]
+	   [ GRANTED BY role_specification ]
+	   [ CASCADE | RESTRICT ]
+	*/
+	StringInfo command = makeStringInfo();
+
+	/* GRANT/REVOKE */
+	if (stmt->is_grant)
+		appendStringInfo(command, "GRANT ");
+	else
+		appendStringInfo(command, "REVOKE ");
+
+	/* privileges [, ...] | ALL */
+	if (stmt->privileges == NULL)
+	{
+		/* ALL */
+		appendStringInfo(command, "ALL ");
+	}
+	else
+	{
+		foreach (lc, stmt->privileges)
+		{
+			AccessPriv *priv = lfirst(lc);
+
+			appendStringInfo(command,
+							 "%s%s ",
+							 priv->priv_name,
+							 lnext_compat(stmt->privileges, lc) != NULL ? "," : "");
```

Comment:
I am not sure what it the proposal here, because format will emit error on the empty char string. Not sure if it is even supported for the C formatting function to accept `''`, I really doubt that it could have numeric equivalent.

Reply:
should be a whitespace. But nevermind, strings work too. But I usually prefer to work with correct types if you are adding a single char.